src/frontend/src/instance.rs

```diff
@@ -212,6 +214,11 @@ impl Instance {
                 self.statement_executor.execute_sql(stmt, query_ctx).await
             }
         };
+
+        if let Some(slow_query_timer) = slow_query_timer {
+            slow_query_timer.stop().await;
```

Comment:
I think "stop" should be done in dropping the timer. The obstacles are the async functions calls, then they are to be moved to spawned tasks. That way (by making slow query logging async) you can avoid slow query logging from affecting the main read workload.